qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -136,94 +139,125 @@ def __init__(self, width: ParameterValueType):
-        super()._set_backend(backend)
+    def set_experiment_options(self, **fields):
+        if "flat_top_widths" in fields:
+            # TODO remove this in Qiskit Experiments 0.6
+            warnings.warn(
+                "'flat_top_widths' argument has been deprecated and will be removed. "
+                "Use 'durations' instead. New variable includes pulse ramps, "
+                "and it cannot include zero in the sequence. "
+                "This argument will be dropped with this warning in Qiskit Experiments 0.6.",
+                DeprecationWarning,
+            )
+        super().set_experiment_options(**fields)
 
-        if self._cr_gate is None:
-            # This falls into CRPulseGate which requires pulse schedule
+    def _get_width(self, duration: ParameterValueType) -> ParameterValueType:
+        """A helper function to get flat top width.
 
-            # Extract control channel index
-            cr_channels = self._backend_data.control_channel(self.physical_qubits)
-            if cr_channels is not None:
-                self._cr_channel = cr_channels[0].index
-            else:
-                warnings.warn(
-                    f"{self._backend_data.name} doesn't provide cr channel mapping. "
-                    "Cannot find proper channel index to play the cross resonance pulse.",
-                    UserWarning,
-                )
+        Args:
+            duration: Cross resonance pulse duration in units of sec.
+
+        Returns:
+            A flat top widths of cross resonacne pulse in units of sec.
+        """
+        timing = BackendTiming(self.backend)
+        backend_dt = timing.dt or 1
+        sigma_sec = self.experiment_options.sigma * backend_dt
 
-            # Extract pulse granularity
-            self._granularity = self._backend_data.granularity
+        return duration - 2 * sigma_sec * self.experiment_options.risefall
 
-        # Extract time resolution, this is anyways required for xvalue conversion
-        self._dt = self._backend_data.dt
-        if self._dt is None:
-            warnings.warn(
-                f"{self._backend_data.name} doesn't provide system time resolution dt. "
-                "Cannot estimate Hamiltonian coefficients in SI units.",
-                UserWarning,
-            )
+    def _get_durations(self) -> np.ndarray:
+        """Return cross resonance pulse durations in units of sec."""
+        opt = self.experiment_options
+
+        if opt.flat_top_widths is not None:
+            # TODO Remove this in Qiskit Experiments 0.6
+            timing = BackendTiming(self.backend)
+            backend_dt = timing.dt or 1
+            widths = np.asarray(opt.flat_top_widths, dtype=float)
+
+            return backend_dt * (widths + 2 * opt.sigma * opt.risefall)
+
+        if opt.durations is None:
+            return np.linspace(opt.durations_min, opt.durations_max, opt.durations_num)
+
+        return np.asarray(opt.durations, dtype=float)
```

Comment:
because user can still provide it in python list. currently there is no practical issue but mixed list/ndarray can sometime cause an issue, e.g. `2 * opt.durations` works with ndarray but not lists.